homeassistant/components/intellidrive/coordinator.py

```diff
@@ -0,0 +1,48 @@
+"""Coordinator for Reisinger Intellidrive."""
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+import reisingerdrive
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_TOKEN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, UPDATE_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ReisingerCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from intellidrive device."""
+
+    config_entry: ConfigEntry
+    device: reisingerdrive.ReisingerSlidingDoorDeviceApi
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize ReisingerCoordinator."""
+        self.device = reisingerdrive.ReisingerSlidingDoorDeviceApi(
+            str(config_entry.data.get(CONF_HOST)),
+            str(config_entry.data.get(CONF_TOKEN)),
```

Comment:
You shouldn't need to do str on host and token. They should come in as str.

Reply:
Done